libnavigation-ui/gradle.properties

```diff
@@ -1,4 +1,4 @@
 POM_ARTIFACT_ID=ui
-POM_NAME=Mapbox Android Navigation UI SDK
-POM_DESCRIPTION=Mapbox Android Navigation UI SDK
-POM_PACKAGING=aar
\ No newline at end of file
+POM_VERSION_NAME=1.0.0-SNAPSHOT
```

Comment:
If we're calling UI lib's version `1.0.0-SNAPSHOT` as well, we can remove all local `POM_VERSION_NAME`s and just keep one variable.

If we'd like to use the `*-v1` scheme, that would require changing the `libnavigation-ui`'s `build.gradle` to not depend on the `project`, but a concrete artifact instead. Or, modifying the POM file generation.

Reply:
> If we're calling UI lib's version `1.0.0-SNAPSHOT` as well, we can remove all local `POM_VERSION_NAME`s and just keep one variable.

I don't think it's a good idea to attach `libnavigation-core` and `libnavigation-ui` version numbers. They could have a separate release pace.

> If we'd like to use the `*-v1` scheme, that would require changing the `libnavigation-ui`'s `build.gradle` to not depend on the `project`, but a concrete artifact instead. Or, modifying the POM file generation.

Yeah, good point! https://github.com/mapbox/mapbox-navigation-android/blob/341e3fa5a0030fda4f424cdabf7cfad05a297708/libnavigation-ui/gradle.properties#L1 Would that require requesting Bintray and Artifactory access for this new artifact? I know that process is tedious 😅

Reply:
> Would that require requesting Bintray and Artifactory access for this new artifact?

Yes, we won't be able to avoid that for public releases.

Reply:
> I don't think it's a good idea to attach libnavigation-core and libnavigation-ui version numbers. They could have a separate release pace.

In this case, we cannot use the `project` dependency.

Reply:
> In this case, we cannot use the `project` dependency.

You mean this https://github.com/mapbox/mapbox-navigation-android/blob/341e3fa5a0030fda4f424cdabf7cfad05a297708/gradle/artifact-settings.gradle#L13? Note that we're using https://github.com/mapbox/mapbox-navigation-android/blob/341e3fa5a0030fda4f424cdabf7cfad05a297708/gradle.properties#L13 across all the core modules e.g. https://github.com/mapbox/mapbox-navigation-android/blob/341e3fa5a0030fda4f424cdabf7cfad05a297708/libnavigation-core/gradle.properties#L2 but not for the UI https://github.com/mapbox/mapbox-navigation-android/blob/341e3fa5a0030fda4f424cdabf7cfad05a297708/libnavigation-ui/gradle.properties#L2 Also, it seems that @abhishek1508 tested and worked fine https://github.com/mapbox/mapbox-navigation-android/pull/2568#issuecomment-597913857 Am I missing something 🤔?

Reply:
No, I mean
https://github.com/mapbox/mapbox-navigation-android/blob/3561a68cc4999bafdfc7168a79c53315177b71e7/libnavigation-ui/build.gradle#L54-L57

Reply:
The project is going to build just fine, but it won't resolve for the end-user if UI and Core lib versions are different.

Reply:
Ohh I see.

How's is that working in https://github.com/mapbox/mapbox-events-android? We're releasing `libtelemetry` and `libcore` separately and `libtelemetry` includes `libcore` https://github.com/mapbox/mapbox-events-android/blob/4cca0cc677fe1a695a984b0e8bd741d1e734a741/libtelemetry/build.gradle#L54 We should do something similar. Would that work?

Reply:
Looking at the release tag, whenever telemetry is released, the tag also specifies the core version (which always was released earlier). Maybe that's how it's done, based on the tag - https://github.com/mapbox/mapbox-events-android/releases.

@harvsu is that the case?

Reply:
@LukasPaczos Thats right! The tag always specifies the core version in the tag. If separate core release is also happening at the current commit, the new version is included in the tag, otherwise, the last released version.
https://github.com/mapbox/mobile-telemetry/blob/master/docs/android-release-process.md